### PR TITLE
Unknown user by default in tests

### PR DESCRIPTION
**Pull request by kevindew**
Follow up to https://github.com/alphagov/content-publisher/pull/258

This doesn't set a creator by default when we build document models via FactoryBot

It creates two new feature tests which test the expectations of a document with a creator and a document without one.

### CODE REVIEW

File: spec/features/viewing_a_document_with_no_creator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  scenario "User finds document without a creator and views it" do
+    given_there_is_a_document_with_no_creator
+    when_i_visit_the_index_page
+    then_i_see_we_dont_know_who_created_the_document
+    when_i_visit_the_document_page
+    then_i_again_see_we_dont_know_who_created_the_document
+  end
+
+  def given_there_is_a_document_with_no_creator
+    @document = create(:document, title: "Excellent document", creator: nil)
```
Comment (benthorner) [inline spec/features/viewing_a_document_with_no_creator_spec.rb]: Is this title important here?

**Comment by benthorner** [comment]
This looks good, although I think there's another piece of potential pain that the feature tests are becoming harder to find as the names diverge.

File: spec/features/viewing_a_document_with_no_creator_spec.rb (hunk 1 above)
Comment (kevindew) [reply spec/features/viewing_a_document_with_no_creator_spec.rb]: It is not, thanks - I've simplified in c82f825

**Comment by kevindew** [comment]
> This looks good, although I think there's another piece of potential pain that the feature tests are becoming harder to find as the names diverge.

Yup I think working out a criteria to to group the feature tests is a looming challenge. Seems tricky as there's not a 1:1 mapping with them and an application concept.

**Pull request merged**